support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/StageStepExecution.java

```diff
@@ -147,6 +143,20 @@ private static synchronized void enter(Run<?,?> r, StepContext context, String n
                 throw new IllegalStateException("the same flow is trying to reënter the stage " + name); // see 'e' with two dots, that's Jesse Glick for you! - KK
             }
         }
+        // Interrupts any build in the holding state
+        if (eager != null && eager) {
+            for (Integer i : stage.holding) {
+                try {
+                    Executor ex = job.getBuildByNumber(i).getExecutor();
```

Comment:
This probably works well enough, though it is rather indirect; nicer to use `FlowExecution.interrupt` if you can get access to the `FlowExecution`, perhaps by keeping a `FlowExectionOwner` in `Stage` (taking care to deal gracefully with null fields from serialized instances from older plugin versions). I am not sure offhand if there is a concrete downside to going through `Executor` like this; the test would need to verify that the interrupted build’s `result` and `CauseOfInterruption` are being set properly.